doc/test_messages_documentation.py

```diff
@@ -0,0 +1,110 @@
+"""Functional tests for the code examples in the messages documentation."""
+
+from collections import Counter
+from pathlib import Path
+from typing import Counter as CounterType
+from typing import List, TextIO, Tuple
+
+import pytest
+
+from pylint import checkers
+from pylint.config.config_initialization import _config_initialization
+from pylint.lint import PyLinter
+from pylint.message.message import Message
+from pylint.testutils.constants import _EXPECTED_RE
+from pylint.testutils.reporter_for_tests import FunctionalTestReporter
+
+MessageCounter = CounterType[Tuple[int, str]]
+
+
+def get_functional_test_files_from_directory(input_dir: Path) -> List[Tuple[str, Path]]:
```

Comment:
We could use https://github.com/PyCQA/pylint/blob/main/pylint/testutils/functional/find_functional_tests.py#L31, right ? ( yield from get_functional_test_files_from_directory if the file is good.py or bad.py ? Maybe we don't even need to check that it's good.py or bad.py)

Reply:
No, that returns `FunctionalTestFile`:
https://github.com/PyCQA/pylint/blob/80838744c6f739fbf2f49bd1bfb70c98a84531aa/pylint/testutils/functional/test_file.py#L45

That class has a lot of additional methods that do not work with the file structure of the messages documentation. Stuff like `self._parse_options()`.
For the purpose of these tests we only need a tuple of test name and path so I don't think it makes much sense to try and change `FunctionalTestFile` to also allow a completely different directory structure.